src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs

```diff
@@ -226,7 +227,16 @@ private void ProcessRequests(ITestRequestManager testRequestManager)
 
                         case MessageType.CancelDiscovery:
                             {
-                                testRequestManager.CancelDiscovery();
+                                // If testhost has old version, we should call old cancel discovery logic
+                                // to be consistent and not create regression issues
+                                if (this.protocolConfig.Version < ObjectModel.Constants.DefaultProtocolConfig.Version)
```

Comment:
Shouldn't this change to the literal version where the change is introduced, that is 6? If the default version changes to 7 in the future, you would be calling the old api for 6 as well.

Reply:
That is correct, will fix it.